### ISSUE TITLE
Deleting and recreating a named server results in lost name in GUI

### ISSUE THREAD

**Issue by PMende** [jupyterhub/jupyterhub#2507]
## Description

I've been working on a POC for my place of work to examine the feasibility of using JupyterHub to serve Jupyter Notebook/Lab servers with custom images containing a Python SDK we're working on.

Recently, I've been working on testing out named servers. In that process, I've discovered that if you delete a named server from the browser GUI, then recreate it in (in any fashion, whether by the REST API or through the GUI), that server will no longer appear listed.

## To reproduce

1. Create a named server:

![image](https://user-images.githubusercontent.com/11052254/55366470-c26c8000-549d-11e9-9fff-f5ff4e7ac5b4.png)

2. Delete it:

![image](https://user-images.githubusercontent.com/11052254/55366831-4b37eb80-549f-11e9-9fec-20d4298e55ab.png)

3. Create it again: `curl -X POST -H "Authorization: token a_very_secret_token" "http://my.host.domain/hub/api/users/pmende/servers/serverA"`

Now the user's Hub Control Panel/Home still no longer lists the server (i.e., it is identical to the image after 2, above), but there is definitely a running pod with the server name:

```
$ kubectl get pods -n jhub
NAME                         READY   STATUS    RESTARTS   AGE
hub-949c864ff-v7dx2          1/1     Running   0          18m
jupyter-pmende-2dserver-41   1/1     Running   0          3m44s
proxy-c88fd6f59-s8k82        1/1     Running   0          18m
```

## Hub creation

`helm upgrade --install jhub jupyterhub/jupyterhub --namespace jhub --version=0.9-8ed2f81 --values config.yaml`

## Contents of `config.yaml`
```
#########################
# Networking Config     #
#########################
proxy:
  secretToken: "mysupersecrettoken"
  service:
    type: NodePort
    nodePorts:
      http: 31212
  chp:
    resources:
      requests:
        memory: 0
        cpu: 0
ingress:
  enabled: true
  hosts:
    - my.host.domain
  rules:
  http:
    - paths: /hub/api
      backend:
        serviceName: hub
        servicePort: 8081

#########################
# Hardware/Image Config #
#########################
singleuser:
  image:
    name: jupyter/scipy-notebook
    tag: 59b402ce701d
  cpu:
    guarantee: 0.25
    limit: 0.5
  memory:
    guarantee: "256M"
    limit: "320M"
  profileList:
    - display_name: "Default"
      description: "0.25 CPU; 256M Ram"
      default: True
    - display_name: "BIG"
      description: "0.5 Whole CPUs, 512M Ram"
      kubespawner_override:
        cpu_guarantee: 0.5
        cpu_limit: 0.75
        mem_guarantee: "512M"
        mem_limit: "640M"

#########################
# Hub Config            #
#########################
hub:
  allowNamedServers: true
  extraConfig: |
    c.JupyterHub.admin_access = True
    c.JupyterHub.api_tokens = {
      "a_very_secret_token": "pmende"
    }

```

**Comment by minrk** [comment]
Migrated issue to jupyterhub. Will need to do some testing. There appear to be two things wrong:

1. when deleting the server, the pod should have been deleted. This *may* be a kubespawner bug, but I'm suspecting it's not. Delete is two actions: stop the server and then delete the entry. It looks like stop may have been skipped. Stop is what deletes the pod.
2. removal from the UI is a different, should be unrelated bug. I'm not sure how that would happen, but thanks for the test case.

**Comment by minrk** [comment]
Turns out it was just the one bug. This is fixed by #2510
